src/main/java/isaac/bastion/listeners/ModeListener.java

```diff
@@ -0,0 +1,155 @@
+		this.settingMan = Bastion.getSettingManager();
+		settingMan.getBsiOverlay().registerListener(new SettingChangeListener<Boolean>() {
+			@Override
+			public void handle(UUID player, PlayerSetting<Boolean> setting, Boolean oldValue, Boolean newValue) {
+				setBsiOverlay(Bukkit.getPlayer(player), newValue);
+			}
+		});
+		settingMan.getBsiLocation().registerListener(new SettingChangeListener<String>() {
+			@Override
+			public void handle(UUID player, PlayerSetting<String> setting, String oldValue, String newValue) {
+				setBsiOverlay(Bukkit.getPlayer(player), settingMan.getBsiOverlay().getValue(player));
+			}
+		});
+	}
+
+	@EventHandler(priority = EventPriority.HIGHEST, ignoreCancelled = true)
+	public void onPlayerJoin(PlayerJoinEvent pje) {
+		if (!settingMan.getBsiOverlay().getValue(pje.getPlayer())) {
+			return;
+		}
+		checkLocationForBastions(pje.getPlayer(), pje.getPlayer().getLocation());
+	}
+
+	@EventHandler(priority = EventPriority.HIGHEST, ignoreCancelled = true)
+	public void onPlayerMove(PlayerMoveEvent pme) {
+		Location from = pme.getFrom();
+		Location to = pme.getTo();
+		if (to == null) {
+			return;
+		}
+
+		if (from.getBlockX() == to.getBlockX() && from.getBlockY() == to.getBlockY()
+				&& from.getBlockZ() == to.getBlockZ() && from.getWorld().equals(to.getWorld())) {
+			// Player didn't move by at least one block
+			return;
+		}
+
+		if (!settingMan.getBsiOverlay().getValue(pme.getPlayer())) {
+			return;
+		}
+
+		checkLocationForBastions(pme.getPlayer(), to);
+	}
+
+	private void setBsiOverlay(Player player, boolean state) {
+		updateDisplaySetting(player, settingMan.getBsiLocation(), state,"",
+				bsiBottomLine, bsiBoard);
+	}
+
+	/**
+	 * This method handles the updating of the scoreboard/action bar
+	 * as to whether the player is in a bastion field or not.
+	 *
+	 * @param player to update bastion status to
+	 * @param location to check for bastion status
+	 */
+	public void checkLocationForBastions(Player player, Location location) {
+		if (player == null || location == null) {
+			return;
+		}
+		Set<BastionBlock> bastionBlocks = Bastion.getBastionManager().getBlockingBastions(location);
+		if (bastionBlocks.isEmpty()) {
+			if (settingMan.getShowNoBastion(player.getUniqueId())) {
+				updateDisplaySetting(player, settingMan.getBsiLocation(), settingMan.getBsiOverlay().getValue(player.getUniqueId()),
+						ChatColor.WHITE + "No Bastion", bsiBottomLine, bsiBoard);
+				return;
+			}
+			bsiBoard.hide(player);
+			bsiBottomLine.removePlayer(player);
+			return;
+		}
+		//We want to check if im in the field of a bastion I dont have perms on
+		//And if so, display enemy bastion
+		//Otherwise we're in a friendly bastion/no bastion
+		for (BastionBlock bastions : bastionBlocks) {
+			if (!bastions.canPlace(player)) {
+				updateDisplaySetting(player, settingMan.getBsiLocation(), settingMan.getBsiOverlay().getValue(player.getUniqueId()),
```

Comment:
This should also show overlap, not just enemy and friendly

Reply:
Basic algo would be like this:
```
boolean hasEnemy = false;
boolean hasFriendly = false;
for (BastionBlock bastions : bastionBlocks) {
			if (bastions.canPlace(player)) {
						hasFriendly = true;
			}
			else {
						hasEnemy = true;
			}
}
if (hasEnemy && hasFriendly) {
			//overlap
}
if (hasEnemy) {
			//only enemy
}
if (hasFriendly) {
			//only friendly
}
```

Additionally it'd be good if you actually differentiated between the different types of bastions. For example an enemy claims bastion would show as allied in your thing, because you can place in it. Then again though that would get really complicated it its supposed to work well with multiple types in general. It'd probably be easiest to just ignore/skip any bastion types for this check, which do not prevent block placements, there should be a method on the bastion which holds its config/type, which has that information